### PR TITLE
Fix bug in mapping due to missing short_id field.

### PR DESCRIPTION
**Pull request by alvacouch**
@aphelionz @martinseul One bug fix of Zhaokun's didn't get into last PR. This corrects a problem with mapping that causes every box to show up when you click on one marker.

### CODE REVIEW

File: hs_core/views/discovery_json_view.py [hunk 1]
```diff
@@ -71,5 +72,5 @@ def form_valid(self, form):
             # return JSON response
 
             return HttpResponse(the_data, content_type='application/json')
-        else:
-            return HttpResponse(json.dumps('[]'), content_type='application/json')
+        # else:
```
Comment (aphelionz) [inline hs_core/views/discovery_json_view.py]: @alvacouch What will happen here if there is no GET data in the request? a 500 error?

Comment (alvacouch) [reply hs_core/views/discovery_json_view.py]: You tell me what should happen. I returned nothing. Zhaokun disagreed.

**Comment by alvacouch** [comment]
@aphelionz We need to rewrite the whole thing as POST because there is a danger of exceedng GET parameter length limits.

**Comment by hydrobuild** [comment]
Build finished.

File: hs_core/views/discovery_json_view.py (hunk 1 above)
Comment (aphelionz) [reply hs_core/views/discovery_json_view.py]: Well we should explicitly return _something_, we can't just let the system decide what gets sent along. I say just uncomment these for now and keep the one-line fix.

**Comment by aphelionz** [comment]
@alvacouch I have performed the edit

**Comment by hydrobuild** [comment]
Build finished.

**Pull request merged**